### PR TITLE
Examples for Method object

### PR DESCRIPTION
**Pull request by mccarthy677**
I copied the five files we discussed, ignoring the method_advanced_user page

### CODE REVIEW

**Review by deepanshs — APPROVED** [review]

File: docs/user_guide/method/writing_custom_methods.rst [hunk 1]
```diff
@@ -0,0 +1,528 @@
...
+======================
+
+The power of mrsimulator lies in the method object. Most time-domain NMR simulation
+software requires users to walk through all aspects of a pulse sequence, but mrsimulator
+preforms calculations in the frequency-domain and only needs descriptions of the spectral
+dimensions being simulated. For a more in-depth discussion on how a method defined
+spectral dimensions, see the :ref:`method_documentation`.
+
+On this page, we will illustrate how to write custom methods by simulating progressively
+more complicated NMR experiments on :math:`RbNO_3`. First lets import the necessary
```
Comment (mgiammar) [inline docs/user_guide/method/writing_custom_methods.rst:14]: Change RbNO3 math directives to :math:`\text{RbNO}_3` so characters are not slanted

File: docs/user_guide/method/writing_custom_methods.rst [hunk 2]
```diff
@@ -0,0 +1,528 @@
...
+    from pprint import pprint
+
+Now we build the :math:`RbNO_3` spin system.
+
+.. plot::
+    :context: close-figs
+
+    site1 = Site(
+        isotope="87Rb",
+        isotropic_chemical_shift=-27.4,  # ppm,
```
Comment (mgiammar) [inline docs/user_guide/method/writing_custom_methods.rst:51]: remove comma after ppm

File: docs/user_guide/method/writing_custom_methods.rst [hunk 3]
```diff
@@ -0,0 +1,528 @@
...
+                    SpectralEvent(transition_query=[{"ch1": {"P": [-1]}}])
+                ]
+            )
+        ]
+    )
+
+The *channels* key holds the nucleus being probed, here rubidium-87. The
+*magnetic_flux_density* key holds the external magnetic field strength in T, and
+*rotor_frequency* holds the rotor frequency in Hz. The *spectral_dimensions* key
+holds a list of SpectralDimension objects defining the spectral grid of the method.
```
Comment (mgiammar) [inline docs/user_guide/method/writing_custom_methods.rst:100]: Add ref to SpectralDimension api

File: docs/user_guide/method/writing_custom_methods.rst [hunk 4]
```diff
@@ -0,0 +1,528 @@
...
+each SpectralDimension object contains a *count* key, defining the number of points
+in that dimension, a *spectral_width* key, containing the spectral width in Hz,
```
Comment (mgiammar) [inline docs/user_guide/method/writing_custom_methods.rst:102]: containing -> defining

File: docs/user_guide/method/writing_custom_methods.rst [hunk 5]
```diff
@@ -0,0 +1,528 @@
...
+    sim.run()
+
+Now, we create a signal processing object to add some exponential line broadening
+to the simulated spectrum and plot the processed dataset.
+
+.. skip: next
+
+.. plot::
+    :context: close-figs
+    :caption: A simulated one-pulse acquire spectrum of :math:`{87}^\text{Rb}` with all sidebands shown (left) and zoomed in plot of the central transition (right).
```
Comment (mgiammar) [inline docs/user_guide/method/writing_custom_methods.rst:124]: Superscript in math directive is backwards

File: docs/user_guide/method/writing_custom_methods.rst [hunk 6]
```diff
@@ -0,0 +1,528 @@
...
+.. simulation, apply our processing, and plot the data.
+
+.. skip: next
+
+.. plot::
+    :context: close-figs
+    :caption: A simulated central-transition selective spectrum of :math:`{87}^\text{Rb}`. The large number of sidebands from the previous simulation have been suppressed.
+
+    sim.methods = [ct_pulseacquire]
+    sim.config.number_of_sidebands = 70  # Reset number of sidebands for efficiency
```
Comment (mgiammar) [inline docs/user_guide/method/writing_custom_methods.rst:185]: Reset -> Reduce

File: docs/user_guide/method/writing_custom_methods.rst [hunk 7]
```diff
@@ -0,0 +1,528 @@
...
+``ct_pulseacquire`` method and re-simulate the spectrum.
+
+.. We simply add this new method to the simulator object, run the
+.. simulation, apply our processing, and plot the data.
+
+.. skip: next
+
+.. plot::
+    :context: close-figs
+    :caption: A simulated central-transition selective spectrum of :math:`{87}^\text{Rb}`. The large number of sidebands from the previous simulation have been suppressed.
```
Comment (mgiammar) [inline docs/user_guide/method/writing_custom_methods.rst:182]: Superscript in math directive is backwards

File: docs/user_guide/method/writing_custom_methods.rst [hunk 8]
```diff
@@ -0,0 +1,528 @@
...
+    )
+
+Now, instead of just one item in the list of spectral dimensions, we
+have two, because 3Q-MAS is a two-dimensional experiment. In the first
+dimension, we are selecting the triple-quantum coherence by specifying a
+transition query of :math:`p=-3` and :math:`d=0`. In the MAS dimension,
+we are selecting the central transition with a transition query of
+:math:`p=-1` and :math:`d=0`.
+
+Again, we add this method to the simulator object, run the simulation, and
```
Comment (mgiammar) [inline docs/user_guide/method/writing_custom_methods.rst:238]: Again, we replace the previous method object, ``ct_pulseaquire``, with ``mqmas``, run the...

File: docs/user_guide/method/writing_custom_methods.rst [hunk 9]
```diff
@@ -0,0 +1,528 @@
...
+equilibrium magnetization to a desired symmetry, we can eliminate the first rotation in
+mrsimulator by defining the first symmetry as :math:`\mathbb{p} = +1`. Our transition symmetry
+pathway now becomes
+
+.. math::
+
+    \mathbb{p}: +1 \xrightarrow[]{\pi} -1
+
+Below is a method object which simulated the Hahn Echo experiment. The MixingEvent defines the
+:math:`\pi` rotation between the two SpectralEvents. We also plot the transition pathways for
```
Comment (mgiammar) [inline docs/user_guide/method/writing_custom_methods.rst:399]: ~~We also plot the transition pathways for~~

**Comment by mgiammar** [comment]
Discussion and examples for creating custom Method objects was moved into PR #181.

**Pull request closed without merging**